lib/src/net/coap_matcher.dart

```diff
@@ -236,10 +236,14 @@ class CoapMatcher implements CoapIMatcher {
     var keyId = CoapKeyId(response.id);
     final keyToken = CoapKeyToken(response.token);
     final exchange = _exchangesByToken[keyToken];
+    final multicast = exchange?.request?.multicast ?? false;
     if (exchange != null) {
       // There is an exchange with the given token
       final prev = _deduplicator!.findPrevious(keyId, exchange);
       if (prev != null) {
+        if (multicast) {
+          return prev;
```

Comment:
Maybe a log line here saying we are returning prev because of multicast processing, may help in debugging this at run time.

Reply:
Good point! Maybe something like this? 

```suggestion
          _log!.info('Matcher - Reuse exchange for multicast processing');
          return prev;
```